### PR TITLE
fix(acceptance-tests): Add TimedRetryPolicy decorator to login_as

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
There are two flaky errors associated with the OrganizationPluginDetailView acceptance test.
1. [ReadTimeout](https://sentry.io/organizations/sentry/issues/?project=2423079&query=is%3Aunresolved+HTTPConnectionPool&statsPeriod=14d
)

Acceptance tests keep failing on `self.login_as` here https://github.com/getsentry/sentry/blob/master/tests/acceptance/test_organization_plugin_detail_view.py#L22
with the following error:
```
ReadTimeoutError: HTTPConnectionPool(host='127.0.0.1', port=49586): Read timed out. (read timeout=<object object at 0x103f815e0>)
```
It is failing to save the cookie on this line:
https://github.com/getsentry/sentry/blob/master/src/sentry/utils/pytest/selenium.py#L394

This error is not exclusive to this acceptance test.

2. [Element does not exist](https://sentry.io/organizations/sentry/issues/1908396608/events/f0bf19d5312a43ca860c18aa9efa7a46/?project=2423079)

where selenium cannot find the form input's css id.

## Solution 
So in this PR, I added a `data-test-id` to the Plugin Configuration Form (bc the test checks for it's existence not contents). So now, selenium looks for the specific `data-test-id`. 

For the `ReadTimeoutError`, I added a TimeoutRetryPolicy decorator on `login_as`.

### CODE REVIEW

**Comment by EvanPurkhiser** [comment]
Will this make acceptance tests significantly slower?

**Comment by billyvg** [comment]
Yeah, won't this add 5 seconds for every test case?

**Comment by NisanthanNanthakumar** [comment]
> Will this make acceptance tests significantly slower?

@EvanPurkhiser it will only run on setup and if there is no cookie saved. But maybe there is a better option?

**Comment by billyvg** [comment]
If we're sure it's `save_session` causing issues, we could make use `TimedRetryPolicy` to retry it?

**Comment by NisanthanNanthakumar** [comment]
> TimedRetryPolicy

@billyvg I think there is a race condition, where we are trying to fetch `self.get("/")` but we continue on to the next operation in the tests. Without the time.sleep, sometimes it passes and sometimes it fails. Will this help with that?

**Comment by wedamija** [comment]
> > Will this make acceptance tests significantly slower?
> 
> @EvanPurkhiser it will only run on setup. But maybe there is a better option?

Setup runs on every test. It looks like the acceptance tests are much slower after this.

**Comment by BYK** [comment]
Wouldn't using [`set_page_load_timeout`](https://selenium-python.readthedocs.io/api.html#selenium.webdriver.remote.webdriver.WebDriver.set_page_load_timeout) solve the issue

>@billyvg I think there is a race condition, where we are trying to fetch self.get("/") but we continue on to the next operation in the tests. Without the time.sleep, sometimes it passes and sometimes it fails. Will this help with that?

Aren't everything here sync? Can you point to a specific line where you think the race condition is?

**Comment by NisanthanNanthakumar** [comment]
> Wouldn't using [`set_page_load_timeout`](https://selenium-python.readthedocs.io/api.html#selenium.webdriver.remote.webdriver.WebDriver.set_page_load_timeout) solve the issue
> 
> > @billyvg I think there is a race condition, where we are trying to fetch self.get("/") but we continue on to the next operation in the tests. Without the time.sleep, sometimes it passes and sometimes it fails. Will this help with that?
> 
> Aren't everything here sync? Can you point to a specific line where you think the race condition is?

Hmm so last week when I was running tests, it would fail during the setup of the test on `self.login_as` when saving the cookie. So I thought we were moving on without letting selenium initialize cookies here: https://github.com/getsentry/sentry/blob/master/src/sentry/utils/pytest/selenium.py#L394.

But today I did a reinstall everything and removed the `time.sleep` and ran it 20 times and it passed each time. So I don't think that was the problem...

So for now, since everything is passing and all the `acceptance` Github Actions are passing on this branch with no changes and now I am not able to reproduce the failure, I'm going to **close this PR**.

**Comment by billyvg** [comment]
@NisanthanNanthakumar This test does fail a bunch on CI:  https://sentry.io/organizations/sentry/discover/results/?field=title&field=transaction&field=environment&field=timestamp&name=AssertionError%3A+assert+False&project=2423079&query=issue.id%3A1908396608&sort=-timestamp&statsPeriod=90d&widths=-1&widths=-1&widths=-1&widths=-1

**Comment by wedamija** [comment]
Acceptance tests still look slower after this change. Not sure if just an unlucky run, but usually each acceptance test is 12-16 min, and now they're all 16-20.

File: src/sentry/testutils/cases.py [hunk 1]
```diff
@@ -178,6 +179,7 @@ def make_request(self, user=None, auth=None, method=None):
 
     # TODO(dcramer): ideally superuser_sso would be False by default, but that would require
     # a lot of tests changing
+    @TimedRetryPolicy.wrap(timeout=5)
```
Comment (wedamija) [inline src/sentry/testutils/cases.py:182]: Does this actually need to be 5 seconds? Would 1 work ok?

Comment (NisanthanNanthakumar) [reply src/sentry/testutils/cases.py:182]: 5 was an arbitrary choice. We can try 1 second and go up from there.

Comment (billyvg) [reply src/sentry/testutils/cases.py:182]: Let's keep it at 5 for now, this shouldn't have too much of an affect on build times as this should only run when `login_as` fails. It's also been hard to reproduce/test due to CI flakyness, but I think we can check to see that these issues drop to near 0 https://sentry.io/organizations/sentry/issues/?project=2423079&query=is%3Aunresolved+HTTPConnectionPool&statsPeriod=14d

**Review by billyvg — APPROVED** [review]

**Pull request merged**